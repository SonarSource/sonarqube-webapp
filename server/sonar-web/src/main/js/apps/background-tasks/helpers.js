/*
 * SonarQube
 * Copyright (C) 2009-2016 SonarSource SA
 * mailto:contact AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
const ONE_SECOND = 1000;
const ONE_MINUTE = 60 * ONE_SECOND;

function format (int, suffix) {
  return `${int}${suffix}`;
}

export function formatDuration (value) {
  if (!value) {
    return '';
  }
  if (value >= ONE_MINUTE) {
    const minutes = Math.round(value / ONE_MINUTE);
    return format(minutes, 'min');
  } else if (value >= ONE_SECOND) {
    const seconds = Math.round(value / ONE_SECOND);
    return format(seconds, 's');
  } else {
    return format(value, 'ms');
  }
}
