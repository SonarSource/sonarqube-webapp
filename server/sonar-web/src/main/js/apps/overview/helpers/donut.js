import d3 from 'd3';
import React from 'react';

let Sector = React.createClass({
  render() {
    let arc = d3.svg.arc()
        .outerRadius(this.props.radius)
        .innerRadius(this.props.radius - this.props.thickness);
    return <path d={arc(this.props.data)} style={{ fill: this.props.fill }}/>;
  }
});

export default React.createClass({
  getDefaultProps() {
    return {
      size: 30,
      thickness: 6
    };
  },

  render() {
    let radius = this.props.size / 2;
    let pie = d3.layout.pie()
        .sort(null)
        .value(d => d.value);
    let data = this.props.data;
    let sectors = pie(data).map((d, i) => {
      return <Sector key={i} data={d} fill={data[i].fill} radius={radius} thickness={this.props.thickness}/>;
    });
    return <svg width={this.props.size} height={this.props.size}>
      <g transform={`translate(${radius}, ${radius})`}>{sectors}</g>
    </svg>;
  }
});
