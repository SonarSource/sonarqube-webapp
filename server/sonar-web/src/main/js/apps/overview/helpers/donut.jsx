import React from 'react';

const Sector = React.createClass({
  render() {
    const arc = d3.svg.arc()
        .outerRadius(this.props.radius)
        .innerRadius(this.props.radius - this.props.thickness);
    return <path d={arc(this.props.data)} style={{ fill: this.props.fill }}></path>;
  }
});

export default React.createClass({
  getDefaultProps() {
    return {
      size: 30,
      thickness: 6
    };
  },

  render() {
    const radius = this.props.size / 2;
    const pie = d3.layout.pie().sort(null)
        .value(d => {
          return d.value
        });
    const data = this.props.data;
    const sectors = pie(data).map((d, i) => {
      return <Sector
          key={i}
          data={d}
          fill={data[i].fill}
          radius={radius}
          thickness={this.props.thickness}/>;
    });
    return (
        <svg width={this.props.size} height={this.props.size}>
          <g transform={`translate(${radius}, ${radius})`}>{sectors}</g>
        </svg>
    );
  }
});
