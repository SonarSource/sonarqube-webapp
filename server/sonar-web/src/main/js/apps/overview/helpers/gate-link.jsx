import React from 'react';

export default React.createClass({
  render() {
    const url = `${baseUrl}/quality_gates/show/${this.props.gate}`;
    return <a href={url}>{this.props.children}</a>;
  }
});
