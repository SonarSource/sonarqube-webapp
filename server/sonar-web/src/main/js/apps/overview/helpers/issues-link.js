import React from 'react';

export default React.createClass({
  render() {
    let params = Object.keys(this.props.params).map((key) => {
          return `${key}=${encodeURIComponent(this.props.params[key])}`;
        }).join('|'),
        url = `${baseUrl}/component_issues/index?id=${encodeURIComponent(this.props.component)}#${params}`;
    return <a href={url}>{this.props.children}</a>;
  }
});
