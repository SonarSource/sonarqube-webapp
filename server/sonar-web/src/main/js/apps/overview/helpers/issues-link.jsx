import React from 'react';

export default React.createClass({
  render: function () {
    var params = Object.keys(this.props.params).map((key) => {
      return `${key}=${encodeURIComponent(this.props.params[key])}`;
    }).join('|');
    var url = `${baseUrl}/component_issues/index?id=${encodeURIComponent(this.props.component)}#${params}`;
    return <a href={url}>{this.props.children}</a>;
  }
});
