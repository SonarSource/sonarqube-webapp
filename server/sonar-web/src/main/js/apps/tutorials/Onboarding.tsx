/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { connect } from 'react-redux';
import handleRequiredAuthentication from '../../app/utils/handleRequiredAuthentication';
import Modal from '../../components/controls/Modal';
import { ResetButtonLink, Button } from '../../components/ui/buttons';
import { translate } from '../../helpers/l10n';
import { CurrentUser, isLoggedIn } from '../../app/types';
import { getCurrentUser } from '../../store/rootReducer';
import './styles.css';

interface OwnProps {
  onFinish: () => void;
  onOpenOrganizationOnboarding: () => void;
  onOpenProjectOnboarding: () => void;
  onOpenTeamOnboarding: () => void;
}

interface StateProps {
  currentUser: CurrentUser;
}

type Props = OwnProps & StateProps;

export class Onboarding extends React.PureComponent<Props> {
  componentDidMount() {
    if (!isLoggedIn(this.props.currentUser)) {
      handleRequiredAuthentication();
    }
  }

  render() {
    if (!isLoggedIn(this.props.currentUser)) {
      return null;
    }

    const header = translate('onboarding.header');
    return (
      <Modal
        contentLabel={header}
        medium={true}
        onRequestClose={this.props.onFinish}
        shouldCloseOnOverlayClick={false}>
        <header className="modal-head">
          <h2>{header}</h2>
        </header>
        <div className="modal-body">
          <p className="spacer-top big-spacer-bottom">
            {translate('onboarding.header.description')}
          </p>
          <ul className="onboarding-choices">
            <li className="text-center">
              <p className="big-spacer-bottom">{translate('onboarding.analyze_public_code')}</p>
              <Button onClick={this.props.onOpenProjectOnboarding}>
                {translate('onboarding.analyze_public_code.button')}
              </Button>
            </li>
            <li className="text-center">
              <p className="big-spacer-bottom">{translate('onboarding.analyze_private_code')}</p>
              <Button onClick={this.props.onOpenOrganizationOnboarding}>
                {translate('onboarding.analyze_private_code.button')}
              </Button>
            </li>
            <li className="text-center">
              <p className="big-spacer-bottom">
                {translate('onboarding.contribute_existing_project')}
              </p>
              <Button onClick={this.props.onOpenTeamOnboarding}>
                {translate('onboarding.contribute_existing_project.button')}
              </Button>
            </li>
          </ul>
        </div>
        <footer className="modal-foot">
          <ResetButtonLink onClick={this.props.onFinish}>{translate('close')}</ResetButtonLink>
        </footer>
      </Modal>
    );
  }
}

const mapStateToProps = (state: any): StateProps => ({ currentUser: getCurrentUser(state) });

export default connect<StateProps, {}, OwnProps>(mapStateToProps)(Onboarding);
