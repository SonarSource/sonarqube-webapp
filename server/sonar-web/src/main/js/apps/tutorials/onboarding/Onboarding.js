/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import TokenStep from './TokenStep';
import OrganizationStep from './OrganizationStep';
import AnalysisStep from './AnalysisStep';
import { translate } from '../../../helpers/l10n';
import handleRequiredAuthentication from '../../../app/utils/handleRequiredAuthentication';
import './styles.css';

type Props = {
  currentUser: { login: string, isLoggedIn: boolean },
  organizationsEnabled: boolean,
  sonarCloud: boolean
};

type State = {
  organization?: string,
  step: string,
  token?: string
};

export default class Onboarding extends React.PureComponent {
  props: Props;
  state: State;

  constructor(props: Props) {
    super(props);
    this.state = { step: props.organizationsEnabled ? 'organization' : 'token' };
  }

  componentDidMount() {
    if (!this.props.currentUser.isLoggedIn) {
      handleRequiredAuthentication();
    }
  }

  handleTokenDone = (token: string) => {
    this.setState({ step: 'analysis', token });
  };

  handleOrganizationDone = (organization: string) => {
    this.setState({ organization, step: 'token' });
  };

  render() {
    if (!this.props.currentUser.isLoggedIn) {
      return null;
    }

    const { organizationsEnabled, sonarCloud } = this.props;
    const { step, token } = this.state;

    let stepNumber = 1;

    return (
      <div className="page page-limited">
        <header className="page-header">
          <h1 className="page-title">
            {translate(sonarCloud ? 'onboarding.header.sonarcloud' : 'onboarding.header')}
          </h1>
          <div className="page-description">
            {translate('onboarding.header.description')}
          </div>
        </header>

        {organizationsEnabled &&
          <OrganizationStep
            currentUser={this.props.currentUser}
            onContinue={this.handleOrganizationDone}
            open={step === 'organization'}
            stepNumber={stepNumber++}
          />}

        <TokenStep
          onContinue={this.handleTokenDone}
          open={step === 'token'}
          stepNumber={stepNumber++}
        />

        <AnalysisStep
          organization={this.state.organization}
          open={step === 'analysis'}
          sonarCloud={sonarCloud}
          stepNumber={stepNumber}
          token={token}
        />
      </div>
    );
  }
}
