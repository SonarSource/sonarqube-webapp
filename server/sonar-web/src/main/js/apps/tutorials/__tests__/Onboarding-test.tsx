/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import { Onboarding } from '../Onboarding';
import { click } from '../../../helpers/testUtils';

it('renders correctly', () => {
  expect(
    shallow(
      <Onboarding
        currentUser={{ isLoggedIn: true }}
        onFinish={jest.fn()}
        onOpenOrganizationOnboarding={jest.fn()}
        onOpenProjectOnboarding={jest.fn()}
        onOpenTeamOnboarding={jest.fn()}
      />
    )
  ).toMatchSnapshot();
});

it('should correctly open the different tutorials', () => {
  const onFinish = jest.fn();
  const onOpenOrganizationOnboarding = jest.fn();
  const onOpenProjectOnboarding = jest.fn();
  const onOpenTeamOnboarding = jest.fn();
  const wrapper = shallow(
    <Onboarding
      currentUser={{ isLoggedIn: true }}
      onFinish={onFinish}
      onOpenOrganizationOnboarding={onOpenOrganizationOnboarding}
      onOpenProjectOnboarding={onOpenProjectOnboarding}
      onOpenTeamOnboarding={onOpenTeamOnboarding}
    />
  );

  click(wrapper.find('ResetButtonLink'));
  expect(onFinish).toHaveBeenCalled();

  wrapper.find('Button').forEach(button => click(button));
  expect(onOpenOrganizationOnboarding).toHaveBeenCalled();
  expect(onOpenProjectOnboarding).toHaveBeenCalled();
  expect(onOpenTeamOnboarding).toHaveBeenCalled();
});
