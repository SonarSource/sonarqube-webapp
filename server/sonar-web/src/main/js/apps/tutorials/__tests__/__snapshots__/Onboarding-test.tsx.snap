// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  contentLabel="onboarding.header"
  medium={true}
  onRequestClose={[MockFunction]}
  shouldCloseOnOverlayClick={false}
>
  <header
    className="modal-head"
  >
    <h2>
      onboarding.header
    </h2>
  </header>
  <div
    className="modal-body"
  >
    <p
      className="spacer-top big-spacer-bottom"
    >
      onboarding.header.description
    </p>
    <ul
      className="onboarding-choices"
    >
      <li
        className="text-center"
      >
        <p
          className="big-spacer-bottom"
        >
          onboarding.analyze_public_code
        </p>
        <Button
          onClick={[MockFunction]}
        >
          onboarding.analyze_public_code.button
        </Button>
      </li>
      <li
        className="text-center"
      >
        <p
          className="big-spacer-bottom"
        >
          onboarding.analyze_private_code
        </p>
        <Button
          onClick={[MockFunction]}
        >
          onboarding.analyze_private_code.button
        </Button>
      </li>
      <li
        className="text-center"
      >
        <p
          className="big-spacer-bottom"
        >
          onboarding.contribute_existing_project
        </p>
        <Button
          onClick={[MockFunction]}
        >
          onboarding.contribute_existing_project.button
        </Button>
      </li>
    </ul>
  </div>
  <footer
    className="modal-foot"
  >
    <ResetButtonLink
      onClick={[MockFunction]}
    >
      close
    </ResetButtonLink>
  </footer>
</Modal>
`;
